app/controllers/users_controller.rb

```diff
@@ -254,6 +261,17 @@ def handle_settings_tab
 
   private
 
+  def set_or_sign_in_user
+    @user = current_user
+
+    if @user
+      authorize @user
+    else
+      flash[:alert] = "You must be logged in to proceed with account deletion."
+      redirect_to new_user_session_path
+    end
+  end
```

Comment:
I have tried rewriting this a handful of times now and have spent a lot of time attempting to move this redirect into `#confirm_destroy` and _out_ of this `before_action` and have been unsuccessful in my attempts. I am open to suggestions if anyone has ideas for how to get this to work within `#confirm_destroy` rather than in the `before_action`, `set_or_sign_in_user`!

Reply:
What happens when you move this logic into that action? Do you run into errors in your server logs, or does it just not work correctly?

At first glance, my thought is that it might be due to multiple redirects happening within the same controller, but I'm not sure if you're getting an error message or not

Reply:
@vaidehijoshi the most common error I get (I've been playing around with the logic within the action itself a lot 😅 ) is a `500` `Pundit::AuthorizationNotPerformedError `:
```
Completed 500 Internal Server Error in 3ms (ActiveRecord: 0.0ms | Allocations: 2366)

Pundit::AuthorizationNotPerformedError - UsersController:
  app/middlewares/set_cookie_domain.rb:13:in `call'

```
When I trace it back in my logs, I see that I _am_ redirected, but cannot get past the error above. It looks like it's an issue stemming from `authorize @user` and I'm unsure of get past it other than through the code I currently have (using the `before_action`).

Reply:
Thank you, @fdoxyz -- works like a charm. 😊